components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesHistoryStorage.kt

```diff
@@ -90,16 +112,16 @@ open class PlacesHistoryStorage(
 
     override suspend fun getVisited(uris: List<String>): List<Boolean> {
         return withContext(readScope.coroutineContext) {
-            handlePlacesExceptions("getVisited", default = uris.map { false }) {
-                places.reader().getVisited(uris)
+            interruptAndHandleExceptions("getVisited", default = uris.map { false }) {
```

Comment:
I don't think it's ideal to interrupt all these calls by default. I think we can be explicit about it, especially as we have a concrete use case where we know we want to interrupt, e.g., In the AwesomeBar when an input change occurred and we're no longer interested in the result of the previous call.

Reply:
So, I would expose the interrupt method instead and call explicitly, as needed. Wdyt?

Reply:
Otherwise, this would also currently introduce undesired behaviour in the `CombinedHistorySuggestionProvider`. There you kick off two `async` calls (`queryHistoryMetadata`, and `getSuggestions`) which will end using the same storage (and reader) instance. Since these calls are not synchronized (and you want them to happen in parallel), the second call would interrupt the first one now, but you want the results of both to combine them.

Reply:
You could combine to a single `async` block or two different readers, but I would've expected a call to interrupt happening in the provider's `onInputChanged` directly.